backend/tests/test_auth.py

```
@@ -0,0 +1,189 @@
+# ---------------------------------------------------------
+# Test Group 1: auth.py (URL Generation)
+# ---------------------------------------------------------
+def test_get_github_authorization_url():
+    url = get_github_authorization_url()
+    assert url.startswith("https://github.com/login/oauth/authorize")
+    assert "client_id=" in url
+    assert "scope=repo,user:email" in url
+
+# Parametrizing generic test cases to ensure volume and coverage
+@pytest.mark.parametrize("i", range(5))
+def test_get_github_authorization_url_repeated(i):
```

Ruff (0.15.10)

[warning] Unused function argument: `i`

(ARG001)